src/vs/workbench/contrib/tasks/browser/abstractTaskService.ts

```diff
@@ -390,24 +389,27 @@ export abstract class AbstractTaskService extends Disposable implements ITaskSer
 			this._storageService.remove(AbstractTaskService.PersistentTasks_Key, StorageScope.WORKSPACE);
 			return true;
 		}
+
 		const tasks = await this.getSavedTasks('persistent');
 		if (!tasks.length) {
 			return true;
 		}
+		const promises = [];
 		for (const task of tasks) {
 			let result;
 			if (ConfiguringTask.is(task)) {
 				const resolved = await this.tryResolveTask(task);
 				if (resolved) {
-					result = await this.run(resolved, undefined, TaskRunSource.Reconnect);
+					result = this.run(resolved, undefined, TaskRunSource.Reconnect);
 				}
 			} else {
-				result = await this.run(task, undefined, TaskRunSource.Reconnect);
-			}
-			if (result?.exitCode === terminalsNotReconnectedExitCode) {
-				this._logService.trace('Terminals were not reconnected');
-				return false;
+				result = this.run(task, undefined, TaskRunSource.Reconnect);
 			}
+			promises.push(result);
+		}
+		const promiseResult = await Promise.all(promises);
+		if (promiseResult.find(p => p?.exitCode === terminalsNotReconnectedExitCode)) {
```

Comment:
note that this case shouldn't happen anymore, so probably safe to remove during debt week